### PR TITLE
Adding ZodZero split keyboard

### PR DESCRIPTION
**Pull request by Aleblazer**
## Description

Adding my split 40% ZodZero split keyboard with Default keymap

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/aleblazer/zodzero/info.json [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    "bootloader": "rp2040",
+    "board": "GENERIC_RP_RP2040",
+    "usb": {
+        "vid": "0xF901",
+        "pid": "0xF911",
+        "device_version": "0.0.1"
+    },
+    "bootmagic": {
+        "matrix": [ 0, 0]
+    },
```
Comment (waffle87) [inline keyboards/aleblazer/zodzero/info.json:16]:
```suggestion
```

File: keyboards/aleblazer/zodzero/keymaps/aleblazer/config.h [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+#ifdef OLED_ENABLE
+  #define OLED_DISPLAY_128X64
+  #define OLED_TIMEOUT 400000
+#endif
+
+#ifdef RGB_MATRIX_ENABLE
+     /* RGB Defines */
+#define RGB_DI_PIN GP0
+#define RGB_MATRIX_LED_COUNT 70
+#define RGBLED_NUM 70
```
Comment (waffle87) [inline keyboards/aleblazer/zodzero/keymaps/aleblazer/config.h:24]:
```suggestion
```

File: keyboards/aleblazer/zodzero/keymaps/aleblazer/config.h [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#define ENABLE_RGB_MATRIX_SPLASH
+#define ENABLE_RGB_MATRIX_MULTISPLASH
+#define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+#endif
```
Comment (waffle87) [inline keyboards/aleblazer/zodzero/keymaps/aleblazer/config.h:73]: I've a feeling this should go at the keyboard level `config.h`? Assuming RGB and OLEDs are meant to be installed. Same goes for other configuration in `rules.mk`, `keymap.c`, etc. Then removed from the other keymaps as well.

File: keyboards/aleblazer/zodzero/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+            tap_code(KC_MS_WH_DOWN);
+        }
+    } else if (index == 1) {
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    }
+    return true;
```
Comment (waffle87) [inline keyboards/aleblazer/zodzero/keymaps/default/keymap.c:380]:
```suggestion
    return false;
```
Although again, this should be moved to the keyboard level using the `encoder_update_kb` function.

File: keyboards/aleblazer/zodzero/readme.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# ZodZero
+
+![ZodZero](https://i.imgur.com/iOvyZLM.jpeg)
```
Comment (waffle87) [inline keyboards/aleblazer/zodzero/readme.md:3]:
```suggestion
![ZodZero](https://i.imgur.com/iOvyZLMh.jpeg)
```

File: keyboards/aleblazer/zodzero/readme.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+* Keyboard Maintainer: [Aleblazer](https://github.com/Aleblazer/), [Discord Link](https://discord.gg/BCSbXwskVt)
+* Hardware Supported: RP2040-Zero
+* Hardware Availability: [Split Logic Keyboards](https://splitlogic.xyz/)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make aleblazer/zodzero:default
+
+Flashing example for this keyboard:
```
Comment (waffle87) [inline keyboards/aleblazer/zodzero/readme.md:15]:
```suggestion
Flashing example for this keyboard:

    make aleblazer/zodzero:default: flash
    
## Bootloader
```

File: keyboards/aleblazer/zodzero/zodzero.h [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  { \
+ { L00,   L01,   L02,   L03,   L04,   L05,   L06,   L07 }, \
+ { L10,   L11,   L12,   L13,   L14,   L15,   L16,   L17 }, \
+ { L20,   L21,   L22,   L23,   L24,   L25,   L26,   L27 }, \
+ { L30,   L31,   L32,   L33,   L34,   L35,   L36,   L37 }, \
+ { R07,   R06,   R05,   R04,   R03,   R02,   R01,   R00 }, \
+ { R17,   R16,   R15,   R14,   R13,   R12,   R11,   R10 }, \
+ { R27,   R26,   R25,   R24,   R23,   R22,   R21,   R20 }, \
+ { R37,   R36,   R35,   R34,   R33,   R32,   R31,   R30 }  \
+}
```
Comment (waffle87) [inline keyboards/aleblazer/zodzero/zodzero.h:32]: This belongs in `info.json`, and this file removed.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/aleblazer/zodzero/keymaps/default/keymap.c
Comment (drashna) [inline keyboards/aleblazer/zodzero/keymaps/default/keymap.c:1]: In general, we want to keep the default keymap simple and plain.  Ideally, it should just be the keymap array, and encoder map array (if it's enabled).

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/aleblazer/zodzero/keymaps/aleblazer/config.h (hunk 3 above)
Comment (Aleblazer) [reply keyboards/aleblazer/zodzero/keymaps/aleblazer/config.h]: Moved this to the keyboard level config.h

File: keyboards/aleblazer/zodzero/keymaps/default/keymap.c (hunk 4 above)
Comment (Aleblazer) [reply keyboards/aleblazer/zodzero/keymaps/default/keymap.c]: Moved to zodzero.h under encoder_update_kb

File: keyboards/aleblazer/zodzero/readme.md (hunk 6 above)
Comment (Aleblazer) [reply keyboards/aleblazer/zodzero/readme.md]: Updated this.

File: keyboards/aleblazer/zodzero/keymaps/aleblazer/keymap.c [hunk 8]
```diff
@@ -0,0 +1,423 @@
...
+
+bool oled_task_user(void) {
+    if (is_keyboard_left()) {
+    render_status();
+    } else {
+        render_anim();  // renders pixelart
+
+        oled_set_cursor(0, 0);                            // sets cursor to (row, column) using charactar spacing (5 rows on 128x32 screen, anything more will overflow back to the top)
+        sprintf(wpm_str, "WPM:%03d", get_current_wpm());  // edit the string to change wwhat shows up, edit %03d to change how many digits show up
+        oled_write(wpm_str, false);                       // writes wpm on top left corner of string
```
[the remark this answers is not in the document]
Comment (Aleblazer) [reply keyboards/aleblazer/zodzero/keymaps/aleblazer/keymap.c]: Removed this keymap.

File: keyboards/aleblazer/zodzero/keymaps/default/config.h [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#define ENABLE_RGB_MATRIX_SPLASH
+#define ENABLE_RGB_MATRIX_MULTISPLASH
+#define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+#endif
```
[the remark this answers is not in the document]
Comment (Aleblazer) [reply keyboards/aleblazer/zodzero/keymaps/default/config.h]: Moved this config to the keyboard level config.h

File: keyboards/aleblazer/zodzero/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,383 @@
...
+            return false;
+        case KC_RAISE:
+            if (record->event.pressed) {
+                layer_on(_RAISE);
+                update_tri_layer(_LOWER, _RAISE, _ADJUST);
+            } else {
+                layer_off(_RAISE);
+                update_tri_layer(_LOWER, _RAISE, _ADJUST);
+            }
+            return false;
```
[the remark this answers is not in the document]
Comment (Aleblazer) [reply keyboards/aleblazer/zodzero/keymaps/default/keymap.c]: Removed this portion of the keymap, using simple layers now.

File: keyboards/aleblazer/zodzero/zodzero.h (hunk 7 above)
Comment (Aleblazer) [reply keyboards/aleblazer/zodzero/zodzero.h]: Updated info.json with the matrix and placement data, removed zodzero.h, and tested with new default keymap.

File: keyboards/aleblazer/zodzero/keymaps/default/rules.mk [hunk 11]
```diff
@@ -0,0 +1,7 @@
+MOUSEKEY_ENABLE  = yes
+EXTRAKEY_ENABLE  = yes
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+WS2812_DRIVER = vendor
+OLED_ENABLE = yes
+OLED_DRIVER = SSD1306
```
[the remark this answers is not in the document]
Comment (Aleblazer) [reply keyboards/aleblazer/zodzero/keymaps/default/rules.mk]: Moved this config to the keyboard level.

File: keyboards/aleblazer/zodzero/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,383 @@
...
+    ),
+  [_ADJUST] = LAYOUT(
+    _______,  KC_F1, KC_F2, KC_F3, KC_F4, KC_F5,                   _______, _______,                   _______, _______, _______, _______, _______, _______,
+    _______,  KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______,  KC_F11, KC_F12, KC_PSCR, KC_SCRL, KC_PAUS, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+    _______,  _______, _______, _______, _______, _______, _______,                                     _______, _______, _______, _______, _______, _______, _______
+    ),
+};
+
+#ifdef OLED_ENABLE
```
[the remark this answers is not in the document]
Comment (Aleblazer) [reply keyboards/aleblazer/zodzero/keymaps/default/keymap.c]: Moved to zodzero.c, and changed oled_task_user to oled_task_kb.

File: keyboards/aleblazer/zodzero/keymaps/default/keymap.c
Comment (Aleblazer) [reply keyboards/aleblazer/zodzero/keymaps/default/keymap.c:1]: Moved all RGB Matrix, Encoder, and OLED code to the keyboard level config.h, rules.mk, and zodzero.c.

**Comment by Aleblazer** [comment]
So to avoid having to rewrite some legacy layer code I typically use, I removed the aleblazer keymap, and on the default, removed the custom layers/keycodes, and just used simple numbered layers. Moved all config that was in the keymap level to the keyboard level, and moved the OLED and Encoder code to the keyboard level zodzero.c. Updated the info.json with the correct positioning, and the matrix data. Deleted zodzero.h.

File: keyboards/aleblazer/zodzero/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+#include <stdio.h>
+bool is_alt_tab_active = false;
+uint16_t alt_tab_timer = 0;
+
+
```
Comment (drashna) [inline keyboards/aleblazer/zodzero/keymaps/default/keymap.c:20]:
```suggestion
```

File: keyboards/aleblazer/zodzero/zodzero.c [hunk 14]
```diff
@@ -0,0 +1,266 @@
...
+}
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    if (is_keyboard_master()) {
+        return OLED_ROTATION_180;
+    }
+    return rotation;
+}
+
+bool oled_task_kb(void) {
```
Comment (drashna) [inline keyboards/aleblazer/zodzero/zodzero.c:257]:
```suggestion
bool oled_task_kb(void) {
    if (!oled_task_user()) {
        return false;
    }
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by Aleblazer** [comment]
@drashna and @waffle87, just checking on if there are any pending reviews needed for this pull request, I believe all request have been addressed.

File: keyboards/aleblazer/zodzero/rules.mk [hunk 15]
```diff
@@ -0,0 +1,9 @@
+SERIAL_DRIVER = vendor
+ENCODER_ENABLE = yes
+MOUSEKEY_ENABLE  = yes
+EXTRAKEY_ENABLE  = yes
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+WS2812_DRIVER = vendor
+OLED_ENABLE = yes
+OLED_DRIVER = SSD1306
```
Comment (drashna) [inline keyboards/aleblazer/zodzero/rules.mk:9]: This is the default, and isn't needed.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/aleblazer/zodzero/readme.md [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+* Keyboard Maintainer: [Aleblazer](https://github.com/Aleblazer/), [Discord Link](https://discord.gg/BCSbXwskVt)
+* Hardware Supported: RP2040-Zero
+* Hardware Availability: [Split Logic Keyboards](https://splitlogic.xyz/)
+
+## Make example for this keyboard (after setting up your build environment):
+
+    make aleblazer/zodzero:default
+
+## Flashing example for this keyboard:
+    make aleblazer/zodzero:default: flash
```
Comment (fauxpark) [inline keyboards/aleblazer/zodzero/readme.md:16]:
```suggestion
Make example for this keyboard (after setting up your build environment):

    make aleblazer/zodzero:default

Flashing example for this keyboard:

    make aleblazer/zodzero:default:flash
```

File: keyboards/aleblazer/zodzero/config.h [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+#define I2C1_SDA_PIN GP14
+
+#ifdef OLED_ENABLE
+  #define OLED_DISPLAY_128X64
+  #define OLED_TIMEOUT 400000
+#endif
+
+#ifdef RGB_MATRIX_ENABLE
+     /* RGB Defines */
+#define WS2812_DI_PIN GP0
```
Comment (fauxpark) [inline keyboards/aleblazer/zodzero/config.h:34]:
```suggestion
```

File: keyboards/aleblazer/zodzero/info.json [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+    "url": "http://splitlogic.xyz/",
+    "maintainer": "Aleblazer",
+    "processor": "RP2040",
+    "bootloader": "rp2040",
+    "board": "GENERIC_RP_RP2040",
+    "usb": {
+        "vid": "0xF901",
+        "pid": "0xF911",
+        "device_version": "0.0.1"
+    },
```
Comment (fauxpark) [inline keyboards/aleblazer/zodzero/info.json:13]:
```suggestion
    },
    "rgb_matrix": {
        "driver": "ws2812"
    },
    "ws2812": {
        "driver": "vendor",
        "pin": "GP0"
    },
```

File: keyboards/aleblazer/zodzero/rules.mk [hunk 19]
```diff
@@ -0,0 +1,9 @@
+SERIAL_DRIVER = vendor
+ENCODER_ENABLE = yes
+MOUSEKEY_ENABLE  = yes
+EXTRAKEY_ENABLE  = yes
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+WS2812_DRIVER = vendor
```
Comment (fauxpark) [inline keyboards/aleblazer/zodzero/rules.mk:7]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**